msc_pygeoapi/loader/metnotes.py

```diff
@@ -0,0 +1,407 @@
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# =================================================================
+
+from datetime import datetime
+import json
+import logging
+import os
+from pathlib import Path
+
+import click
+from elasticsearch import logger as elastic_logger
+
+from msc_pygeoapi import cli_options
+from msc_pygeoapi.connector.elasticsearch_ import ElasticsearchConnector
+from msc_pygeoapi.env import (
+    MSC_PYGEOAPI_CACHEDIR,
+    MSC_PYGEOAPI_LOGGING_LOGLEVEL,
+)
+from msc_pygeoapi.loader.base import BaseLoader
+from msc_pygeoapi.util import (
+    check_es_indexes_to_delete,
+    configure_es_connection
+)
+
+LOGGER = logging.getLogger(__name__)
+elastic_logger.setLevel(getattr(logging, MSC_PYGEOAPI_LOGGING_LOGLEVEL))
+
+DATETIME_FORMAT = '%Y%m%dT%H%M%S.%fZ'
+
+# cleanup settings
+DAYS_TO_KEEP = 7
+
+INDEX_BASENAME = 'metnotes.'
+
+MAPPINGS = {
+        'properties': {
+            'geometry': {
+                'type': 'geo_shape'
+            },
+            'properties': {
+                'properties': {
+                    'id': {
+                        'type': 'text',
+                        'fields': {
+                            'raw': {
+                                'type': 'keyword'
+                            }
+                        }
+                    },
+                    'aors': {
+                        'type': 'keyword',
+                        'index': 'true'
+                    },
+                    'type_id': {
+                        'type': 'text',
+                        'fields': {
+                            'raw': {
+                                'type': 'keyword'
+                            }
+                        }
+                    },
+                    'publication_version': {
+                        'type': 'integer',
+                    },
+                    'start_datetime': {
+                        'type': 'date',
+                        'format': "YYYY-MM-DD'T'HH:mm:ss.SSS'Z'"
+                    },
+                    'end_datetime': {
+                        'type': 'date',
+                        'format': "YYYY-MM-DD'T'HH:mm:ss.SSS'Z'"
+                    },
+                    'expiration_datetime': {
+                        'type': 'date',
+                        'format': "YYYY-MM-DD'T'HH:mm:ss.SSS'Z'"
+                    },
+                    'publication_datetime': {
+                        'type': 'date',
+                        'format': "YYYY-MM-DD'T'HH:mm:ss.SSS'Z'"
+                    },
+                    'metnote_status': {
+                        'type': 'text',
+                        'fields': {
+                            'raw': {
+                                'type': 'keyword'
+                            }
+                        }
+                    },
+                    'filename': {
```

Comment:
What does an example `filename` look like?

Reply:
sure, filename is the filename for the metnote file on HPFX, eg: